### ISSUE TITLE
[FEATURE] Excluded environments for snippets

### ISSUE THREAD

**Issue by BobVarioa** [artisticat1/obsidian-latex-suite#248]
# Description of the Problem
Its annoying when in environments where the syntax is different (e.g. ce, pu) the default snippets still run.  

# Description of the Solution
Either something like the current auto-fraction options to disable all snippets from running in certain environments or a new parameter for each snippet that lets you exclude (/ whitelist) certain environments.

**Comment by Comprehensive-Jason** [comment]
I agree, this would be a great feature. For example, using mhchem's arrows is very annoying since the plugin keeps autocompleting math inside `\ce{}`. I have to type everything with spaces to stop the autocomplete.

https://github.com/artisticat1/obsidian-latex-suite/assets/68677082/3e94fec5-df6f-43c5-bd35-944b743c800f

**Comment by Comprehensive-Jason** [comment]
@BobVarioa I'm thinking of writing a PR for this by simply extending the auto-fraction options.

I will add a blacklist under **Advanced Snippet Settings**. Any environment written in the blacklist will have all snippets disabled unless the snippet has an `e` option added. 

The limitation with this approach is that you can't select specific snippets for specific environments. For instance, if you have certain snippets that you want to run in `\ce{ }` but not in `\physics{ }` and vice versa, there is no way to specifically select for each.

**Comment by artisticat1** [comment]
Excluded environments are actually already implemented in code ([see here](https://github.com/search?q=repo%3Aartisticat1%2Fobsidian-latex-suite%20exclusions&type=code)), though currently the list of exclusions is fixed - there is no way for users to define custom exclusions for their snippets.

What do you think about exposing this functionality to the user through an interface like this (as opposed to a dialog box)?

```typescript
{trigger: "->", replacement: "\\to", options: "mA", excluding: [["\ce{", "}"]]}
```

**Comment by Comprehensive-Jason** [comment]
@artisticat1 This would be perfect! That puts all the customization in one place, which is way simpler.

**Comment by sylph520** [comment]
@artisticat1 I encountered recently that the  i want to enter inline code `mkdir folder_A`, but the mk is automatically fracted to `$$` after I input `mk`,  I found a setting Excluded Environment (seems related), but no detailed notes, so I tried to add a line `["`", "`"]` or `["\'", "\'"]`, but still no luck, it still auto-fracted. Would you tell me how to set this configuration. Thanks.

**Comment by Comprehensive-Jason** [comment]
@sylph520 I don't think this feature has been implemented yet, though the bones are already there as artisticat said. We're waiting on them to add the feature.

**Comment by superle3** [comment]
> Excluded environments are actually already implemented in code (see here), though currently the list of exclusions is fixed - there is no way for users to define custom exclusions for their snippets.

It runs JavaScript and in the replacement function you can change the excluded environment :) as `this` is binded to the snippet object.

Am still waiting to implement it cause wanted to first focus on performance and the current api might be too powerfull for what usecases it needs to fulfil that I can't optimize it properly. If it just needs to detect if it's inside a certain macro than this is too much.
